packages/ember-simple-auth/tests/helpers/create-adaptive-store.js

```diff
@@ -1,25 +1,13 @@
 import Adaptive from 'ember-simple-auth/session-stores/adaptive';
-import createCookieStore from './create-cookie-store';
-import { merge, assign as emberAssign } from '@ember/polyfills';
-
-const assign = emberAssign || merge;
 
 export default function createAdaptiveStore(
   cookiesService,
   options = {},
-  props = {}
+  owner
 ) {
-  let cookieStore = createCookieStore(
-    cookiesService,
-    assign({}, options, { _isFastboot: false })
-  );
-  props._createStore = function() {
-    cookieStore.on('sessionDataUpdated', data => {
-      this.trigger('sessionDataUpdated', data);
-    });
-
-    return cookieStore;
-  };
+  owner.register('session-store:adaptive', Adaptive.extend(Object.assign({
+    _isLocalStorageAvailable: false,
+  }, options)));
 
-  return Adaptive.extend(props).create(options);
+  return owner.lookup('session-store:adaptive');
 }
```

Comment:
Is this function still necessary? I think it could be replaced in tests with

```js
const store = owner.lookup('session-store:adaptive');
store.setProperties(options);
```

?

Reply:
No because we need to set `isLocalStorageAvailable` before `init` where the `session-store:local-storage` is initialized. So the only way to do so is extend the class and register.

It's also a part of the reason why I moved `isLocalStorageAvailable` as computed property, so I can override it and read the value on the init. But this instead could potenitally be handled by checking if `'_isLocalStorageAvailable in this` instead of `this.hasOwnProperty('_isLocalStorageAvailable')` but I thought that computed property is a better way.